app/src/sharedTest/java/org/oppia/app/testing/InputInteractionViewTestActivityTest.kt

```diff
@@ -673,4 +673,146 @@ class InputInteractionViewTestActivityTest {
+        .isEqualTo(listOf(1, 2, 3))
+    }
+  }
+
+  @Test
+  @Ignore("Landscape not properly supported") // TODO(#56): Reenable once landscape is supported.
+  fun testRatioInputInteractionView_withInputtedText_onConfigurationChange_hasCorrectPendingAnswer() { // ktlint-disable max-line-length
+    val activityScenario = ActivityScenario.launch(
+      InputInteractionViewTestActivity::class.java
+    )
+    onView(withId(R.id.test_ratio_input_interaction_view))
+      .perform(
+        typeText(
+          "1:2"
+        )
+      )
+    activityScenario.onActivity { activity ->
+      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+    }
+    onView(withId(R.id.test_ratio_input_interaction_view)).check(matches(isDisplayed()))
+      .check(matches(withText("1:2")))
+  }
+
+  @Test
+  fun testratioInputInteractionView_withInputtedTwoColonsTogether_colonsTogetherFormatErrorIsDisplayed() { // ktlint-disable max-line-length
+    ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_ratio_input_interaction_view))
+      .perform(
+        typeText(
+          "1::2"
+        )
+      )
+    onView(withId(R.id.ratio_input_error))
+      .check(
+        matches(
+          withText(
+            R.string.ratio_error_invalid_colons
+          )
+        )
+      )
+  }
+
+  @Test
+  fun testRatioInputInteractionView_withInputtedInvalidNumberOfColons_numberFormatErrorIsDisplayed() { // ktlint-disable max-line-length
+    ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_ratio_input_interaction_view))
+      .perform(
+        typeText(
+          "1:2:3:4:"
+        )
+      )
+    onView(withId(R.id.ratio_input_error))
+      .check(
+        matches(
+          withText(
+            R.string.ratio_error_invalid_colons
+          )
+        )
+      )
+  }
+
+  @Test
+  fun testRatioInputInteractionView_withInputtedSpacesBetweenComponents_hasCorrectPendingAnswer() {
+    ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_ratio_input_interaction_view))
+      .perform(
+        typeText(
+          "1   : 2 : 3 : 4 "
+        )
+      )
+    onView(withId(R.id.test_ratio_input_interaction_view)).check(matches(isDisplayed()))
+      .check(matches(withText("1:2:3:4")))
+  }
+
+  @Test
+  fun testRatioInputInteractionView_withInputtedSpacesBetweenNumbers_numberFormatErrorIsDisplayed() { // ktlint-disable max-line-length
+    ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_ratio_input_interaction_view))
+      .perform(
+        typeText(
+          "1:2  3:4"
+        )
+      )
+    onView(withId(R.id.ratio_input_error))
+      .check(
+        matches(
+          withText(
+            R.string.ratio_error_invalid_format
+          )
+        )
+      )
+  }
+
+  @Test
```

Comment:
I suggest also testing a ratio with a fraction in it, and submit time errors.

Reply:
done

Reply:
To be clear: we should test all of the real-time & submit time errors here, as well.

Also, did you add any submit-time errors? I only see real-time.

Reply:
for the activity I could only check the real-time error as in this activity there's no button which would behave as a submit button, in order to check that i probably need to check that inside state fragment tests.

Reply:
Added

Reply:
@aggarwalpulkit596 I don't see a test for the invalid number of components. Please make sure _all_ real-time & submit-time errors are covered here.

Reply:
done